integration/mocha-tests/README.md

```diff
@@ -0,0 +1,111 @@
+```
+
+There are scenarios where a test might not be compatible with one of the browsers or OS. In that case, the user can provide a custom config with an updated clients array. `sample_test.config.json` is a sample test config already provided. 
+The following command can be used to run a browser compatibility test with a custom config:
+
+```bash
+npm run test -- --test-name AudioTest --host saucelabs --test-type browser-compatibility --config browserCompatibilityTest/desktop/sample_test.config.json
+```
+
+### Running Browser Compatibility Tests on AWS Device Farm
+To run the tests on Device Farm, you will need an AWS account. Once you have an AWS account, you will need to set up two environment variables that will allow the AWS SDK to authenticate to your AWS account.
+```bash
+export AWS_ACCESS_KEY_ID=<YOUR_ACCESS_KEY_ID>
+export AWS_SECRET_ACCESS_KEY=<YOUR_SECRET_ACCESS_KEY>
+```
+
+You will need to create a Device Farm project in the AWS account that you are planning to use. There are two types of Device Farm projects that can be created: `Mobile Device Testing` and `Desktop Browser Testing`. For this section, we will focus mainly on desktop browser testing.
+After the project is created, you will need to set up a `PROJECT_ARN` as an environment variable. The project ARN is used by the Device Farm API to identify the project to create test sessions inside.
+```bash
+export PROJECT_ARN=<YOUR_PROJECT_ARN>
+```
+
+The following command can be used to run a browser compatibility test with default settings on Device Farm:
+```bash
+npm run test -- --test-name AudioTest --host devicefarm --test-type browser-compatibility
+```
+
+Like Sauce Labs, Device Farm can run browser compatibility tests with a custom config:
+```bash
+npm run test -- --test-name AudioTest --host devicefarm --test-type browser-compatibility --config browserCompatibilityTest/desktop/sample_test.config.json
+```
+
+## Logging
+This section will go over the two distinct ways of logging:
+1. Use `logger.log` for inline logging
+2. Use `logger.pushLogs` for buffered logging
+
+You can use `logger.log` as a default logger with enhanced functionalities. `logger.log` supports two additional features:
+- `logger.log` takes one of the four log levels: `LogLevel.INFO`, `LogLevel.WARN`, `LogLevel.ERROR`, and `LogLevel.SUCCESS`.
+- `logger.log` outputs messages to the console in different colors based on the log level.
+
+Integration tests use `mocha` as their test framework. Mocha prints the test results and some limited logs to the console automatically. Any console logs inside `it` will print before the mocha logs.
+This makes debugging hard as the logs seem out of sync and random. To get around this issue, buffered logging was added. Inside of `it` blocks, `logger.pushLogs` can be used to add logs to the buffer and a call to `logger.printLogs` inside of the `afterEach` hook will print the logs in the desired order.
```

Comment:
Then should this be the default then? Why do we want dev to make decision if one is clearly better to avoid issue?